resteasy-jaxrs/src/main/java/org/jboss/resteasy/client/exception/ResteasyClientErrorException.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2012, 2017 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.jboss.resteasy.client.exception;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * A base runtime application exception indicating a client request error
+ * (HTTP {@code 4xx} status codes).
+ *
+ * @author Marek Potociar
+ * @since 2.0
+ */
+public class ResteasyClientErrorException extends ResteasyWebApplicationException {
+
+    private static final long serialVersionUID = -4101970664444907990L;
+
```

Comment:
Is the UID here actually properly generated? It seems to match the one of JAX-RS ClientErrorException. Please double check the UIDs for all other exceptions being added in this PR.

Reply:
Good catch - I didn't think about that. It turns out that most of them remain unchanged, but a few are different. Done.